bin/propolis-utils/src/bin/cpuid-gen.rs

```diff
@@ -261,15 +262,125 @@ fn print_toml(results: &BTreeMap<CpuidKey, Cpuid>) {
     }
 }
 
+fn print_json(results: &BTreeMap<CpuidKey, Cpuid>) {
+    use propolis_api_types::instance_spec::components::board::{
+        Cpuid, CpuidEntry,
+    };
+
+    let vendor = {
+        use propolis_api_types::instance_spec::{CpuidValues, CpuidVendor};
+        match results.get(&CpuidKey::Leaf(0)) {
+            None => {
+                eprintln!("no result for leaf 0, selecting default vendor");
+                CpuidVendor::Amd
+            }
+            Some(values) => {
+                let values = CpuidValues {
+                    eax: values.eax,
+                    ebx: values.ebx,
+                    ecx: values.ecx,
+                    edx: values.edx,
+                };
+
+                match CpuidVendor::try_from(values) {
+                    Err(_) => {
+                        eprintln!(
+                            "vendor in leaf 0 values ({values:?}) \
+                        not recognized, selecting default vendor"
+                        );
+                        CpuidVendor::Amd
+                    }
+                    Ok(v) => v,
+                }
+            }
+        }
+    };
+
+    // Returns `true` if `key` has subleaf data and its leaf index is `leaf`.
+    fn key_matches_leaf_and_has_subleaves(key: &CpuidKey, leaf: u32) -> bool {
+        let CpuidKey::SubLeaf(l, _) = key else {
+            return false;
+        };
```

Comment:
style nit q:

Could this be worded as
```rust
match key {
    CpuidKey::SubLeaf(l, _) if *l == leaf => true,
    _ => false,
}
```
and then be inlined into the `any()` call below?

Reply:
We can go one better and use `matches!` (I didn't realize at first that it let you append a conditional, but in retrospect it makes sense that it would). Fixed in 41fc83815d2d.